src/insights_client/tests/test_metrics.py

```diff
@@ -285,3 +285,130 @@ def test_metrics_post_event_proxy(post, config_file_factory, rhsm_config_file_fa
         proxies={"https": "http://user:password@localhost:3128"},
     )
 
+@patch("insights_client.metrics.os.getenv")
+def test_is_offline(os_getenv, config_file_factory):
```

Comment:
Pytest has _monkeypatch_ that can be used to patch both the environment and the args.

```suggestion
def test_is_offline(monkeypatch, config_file_factory):
```

Reply:
```suggestion
@patch("insights_client.metrics.os.getenv")
def test_is_offline(os_getenv, raw_config_file_factory):
```